types/google-protobuf/index.d.ts

```diff
@@ -226,6 +226,8 @@ type BinaryReadReader = (msg: any, binaryReader: BinaryReader) => void;
 
 type BinaryRead = (msg: any, reader: BinaryReadReader) => void;
 
+type BinaryReadCallback = (value: any, binaryReader: BinaryReader) => any;
```

Comment:
To match the old types, it seems like it should be generic?
```ts
type BinaryReadCallback<V> = (value: V, binaryReader: BinaryReader) => any;
```

Reply:
Makes this sense for the BinaryWriteCallback as well?
```typescript
type BinaryWriteCallback<V> = (value: V, binaryWriter: BinaryWriter) => void;
```

Reply:
To get any benefit from that, it looks like you'd have to make `BinaryWriter` generic, which seems like a separate change.

Reply:
I've extended the test again by adding more complex map where the key is still a number but the value another protobuf message. I want to be sure that the typings of BinaryWriter, BinaryReader and their callbacks are really correct.

And voilà, with the additional test, the Typescript compiler fails again, there is still something wrong in the typings of google-protobuf. I try to solve this issue and update this PR.

Reply:
Ok, it looks good now. All callbacks are using `any` for value. If you try to change this to generic callbacks you end up with a huge change. If it is OK for you, I would prefer to leave it as it is and to use `any` for the value. Maybe a further review of this topic makes sense but this should be handled in another PR / discussion.